packages/product-json-to-csv/src/map-product-data.js

```diff
@@ -74,92 +106,257 @@ export default class ProductMapping {
     return productWithVariants
   }
 
-  _mapProperties(product: SingleVarPerProduct): MappedProduct {
-    return reduce(
+  static _mapSearchKeywords(value: Object) {
+    const keywords: any = {}
+    Object.keys(value).forEach((language: string) => {
+      const standard = []
+      const whitespace = []
+      value[language].forEach((keyWord: Object) => {
+        if (!keyWord.suggestTokenizer) standard.push(keyWord.text)
+        else if (keyWord.suggestTokenizer.type === 'whitespace')
+          whitespace.push(keyWord.text.replace(/ /g, ' | '))
+      })
+      keywords[language] = [...standard, ...whitespace].join(';')
+    })
+    return keywords
+  }
+
+  _mapVariantProperties(variant: Variant, productType: ?ProductType): Object {
+    const mappedVariant: Object = _.reduce(
+      variant,
+      (acc: Object, value: any, property: string): Object => {
+        acc[property] = this._mapVariantProperty(property, value)
+        return acc
+      },
+      {}
+    )
+
+    // complete missing attributes
+    if (productType && productType.attributes)
+      productType.attributes.forEach((attribute: Object) => {
+        const mappedAttributes = mappedVariant.attributes || {}
+        if (_.isNil(mappedAttributes[attribute.name]))
+          mappedAttributes[attribute.name] = ''
+      })
+
+    return mappedVariant
+  }
+
+  _mapVariantProperty(property: string, value: any): any {
+    switch (property) {
+      case 'attributes':
+        return this._mapAttributes(value)
+      case 'prices':
+        return this._mapPrices(value)
+      case 'images':
+        return this._mapImages(value)
+      case 'availability':
+        return value.availableQuantity
+      case 'id':
+      case 'sku':
+      case 'key':
+        return value
+      default:
+        return undefined
+    }
+  }
+
+  _mapPrices(prices: Array<Price>): string {
+    return prices
+      .map((price: Price) => ProductMapping._mapPrice(price))
+      .join(this.multiValDel)
+  }
+
+  static _mapPrice(price: Price): string {
+    // Full price:
+    // 'country-currencyCode centAmount|discounted.centAmount customerGroup.name#channel.key$validFrom~validUntil'
+    let priceString: string = ''
+
+    if (price.country) priceString += `${price.country}-`
```

Comment:
we do use `oneLineTrim`, which would add to more composition of text all thanks to tag-template

Reply:
Updated here 62828bc